base/primitive_thread_pool.hpp

```diff
@@ -0,0 +1,103 @@
+// This file contains PrimitiveThreadPool class.
+#pragma once
+
+#include "base/assert.hpp"
+#include "base/thread_utils.hpp"
+
+#include <atomic>
+#include <functional>
+#include <future>
+#include <memory>
+#include <queue>
+#include <thread>
+
+namespace threads
+{
+// PrimitiveThreadPool is needed for easy parallelization of tasks.
+// PrimitiveThreadPool can accept tasks that return result as std::future.
+// When the destructor is called, all threads will join.
+//
+// Usage example:
+// size_t threadCount = 4;
+// std::atomic<size_t> counter = 0;
+// {
+//   threads::PrimitiveThreadPool threadPool(threadCount);
+//   for (size_t i = 0; i < threadCount; ++i)
+//   {
+//     threadPool.Submit([&]() {
+//       std::this_thread::sleep_for(std::chrono::milliseconds(1));
+//       ++counter;
+//     });
+//   }
+// }
+// TEST_EQUAL(threadCount, counter, ());
+//
+class PrimitiveThreadPool
+{
+public:
+  using FuntionType = FunctionWrapper;
+  using Threads = std::vector<std::thread>;
```

Comment:
Видимо, надо добавить
```
*NOTE* PrimitiveThreadPool works with std::thread instead of SimpleThread and therefore should not be used when the JVM is needed.
```

?

Reply:
++